src/app/zap-templates/templates/app/attribute-size-src.zapt

```diff
@@ -14,15 +14,36 @@ using namespace chip::app::List;
 // and data is undefined.
 constexpr uint16_t kSizeLengthInBytes = 2u;
 
+#if BIGENDIAN_CPU
+uint8_t *reverseMemcpy(uint8_t *dest, const uint8_t *src, size_t len)
+{
+    uint8_t *d = dest + len - 1;
+    const uint8_t *s = src;
+    while (len--)
+    {
+        *d-- = *s++;
+    }
+    return dest;
+}
+#endif
```

Comment:
```suggestion
#define copyToNativeEndian(dest, src, len) reverseMemcpy(dest, src, len)
#else // BIGENDIAN_CPU
#define copyToNativeEndian(dest, src, len) memmove(dest, src, len)
#endif // BIGENDIAN_CPU
```
and then have no other ifdefs; just use `copyToNativeEndian` in the places where that's what we are doing.